web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -248,6 +262,37 @@ public int size() {
         return tables.size();
     }
 
+    /**
+     * An array of all the key columns that the tables are partitioned by.
+     *
+     * @return Array of Column
+     */
+    @JsProperty
+    public Column[] getKeyColumns() {
+        return keyColumns;
+    }
+
+    /**
+     * An array of the columns in the tables that can be retrieved from this partitioned table, including both key and
+     * non-key columns.
+     *
+     * @return Array of Column
+     */
+    @JsProperty
+    public Column[] getColumns() {
+        return columns;
+    }
+
+    /**
+     * A Table object containing all currently known keys used for this partitioned table.
+     *
+     * @return Table
+     */
+    @JsProperty
+    public JsTable getKeyTable() {
```

Comment:
This is probably not safe to return - what are you trying to do with this as an api consumer?

Reply:
The new UI being implemented for Partitioned Tables ([spec](https://github.com/deephaven/web-client-ui/issues/1143)) need to be able to display the table containing all the possible keys.